codespell_lib/data/dictionary.txt

```diff
@@ -1236,6 +1236,10 @@ algorithemically->algorithmically
 algorithems->algorithms
 algorithic->algorithmic
 algorithically->algorithmically
+algorithim->algorithm
+algorithimes->algorithms
+algorithimical->algorithmical
```

Comment:
The suggested word `algorithmical` is not found in major dictionaries (Eg. Oxford Learners, Merriam-Webster, etc).
If fact, the suggested word `algorithmical` itself is flagged as a misspell in `dictionary.txt`, this also explained the PR checks failure. You might want to change to `algorithimic->algorithmic`